dapp/src/components/page/project/CommitHistory.jsx

```diff
@@ -1,46 +1,49 @@
-import React from "react";
-import { useState, useEffect } from "react";
+import React, { useState, useEffect } from "react";
 import CommitRecord from "./CommitRecord.jsx";
 import { formatDate } from "../../../utils/formatTimeFunctions.ts";
 import { projectInfoLoaded, latestCommit } from "../../../utils/store.js";
-import {
-  loadProjectRepoInfo,
-  loadConfigData,
-} from "../../../service/StateService.ts";
-import { getCommitHistory } from "../../../service/GithubService.ts";
+import { loadProjectRepoInfo, loadConfigData } from "../../../service/StateService.ts";
 import { useStore } from "@nanostores/react";
 import Pagination from "components/utils/Pagination.tsx";
+import axios from "axios";
 
 const CommitHistory = () => {
   const isProjectInfoLoaded = useStore(projectInfoLoaded);
   const [commitHistory, setCommitHistory] = useState([]);
   const [authors, setAuthors] = useState([]);
   const [currentPage, setCurrentPage] = useState(1);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
 
   const fetchCommitHistory = async (page = 1) => {
     const projectRepoInfo = loadProjectRepoInfo();
     if (projectRepoInfo?.author && projectRepoInfo?.repository) {
-      const history = await getCommitHistory(
-        projectRepoInfo.author,
-        projectRepoInfo.repository,
-        page,
-      );
-      setCommitHistory(history);
-      setCurrentPage(page);
+      try {
+        setLoading(true);
+        setError("");
+        console.log("author", projectRepoInfo.author)
+        console.log("repo", projectRepoInfo.repository)
+        const response = await axios.get(`/api/commits?username=${projectRepoInfo.author}&repo=${projectRepoInfo.repository}`)
```

Comment:
To be clear, I don't want to have a backend API. Fetching the repo has to happen client side and be cached there.